packages/oats-runtime/test/make-from-reflection.spec.ts

```diff
@@ -253,13 +253,12 @@ describe('string', () => {
     );
   });
 
-  it('accepts if format is not defined', () => {
```

Comment:
what’s the reasoning behind this? why throw errors?

Reply:
I would kinda ask the same thing. format is kinda unregulated -- anything goes there. Throwing on unknown seems kinda onerous.

Reply:
we want to rely more on oats for request validation. So we're gonna use custom formats ("isoDate", "email", "resourceId" etc). Now without ensuring the format is defined we can easily lose validation (without knowing it!) after refactoring (e.g. it is as simple as registering a format after running `runtime.server.createHandlerFactory()`) or just by making a typo.

Reply:
It's not a big breaking change. So now if you have an unregistered format, `runtime.server.createHandlerFactory()` will throw (it is not nice admittedly bc the error is in runtime). So the options in this case is either to remove `format` from schema (it is quite useless without validator - can be moved to description or summary) or registering it. I'll add more informative exception.

Reply:
It kind of does break things and I'm not convinced that requiring format is important enough to offset the cost.

Openapi 3 is way more lax on its handling of format

> Tools that do not recognize a specific format MAY default back to the type alone, as if the format is not specified

bunch of the string validation cases can be dealt with by `pattern`, would those work for you instead of having to define formats?

Reply:
some of our formats are more complicated than just using regex (and regexp for things like email/isoDate is also not that trivial). Besides this functionality `registerFormat()` was added so we can use it (and we want to!).
I'm wondering, what's the use case of having a format if there's no handler for it?
If it is really breaking, we can introduce some `assertFormatsRegistered` option somewhere (any ideas where?) to disable this assertion. We can even default it to `false`.

Reply:
could we have the format assertion as a separate discussion (and pr)? I think we, or at least I, agree on having the number and string parsers added to the query runtime checking so moving that forward is likely something that would benefit you even without the formatters.

Reply:
i thought we agreed on a driver option, but i can add in the next pr, sure.